library/Vanilla/Authenticator/SSOAuthenticator.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Authenticator;
+
+use Garden\Web\RequestInterface;
+use Vanilla\Models\SSOInfo;
+
+abstract class SSOAuthenticator extends Authenticator {
+    /**
+     * Tells whether the data returned by this authenticator is authoritative or not.
+     * User info/roles can only be synchronized by trusted authenticators.
+     *
+     * @var bool
+     */
+    private $isTrusted = false;
+
+    /**
+     * Authenticator constructor.
+     *
+     * @param string $authenticatorID Currently maps to "UserAuthenticationProvider.AuthenticationKey".
+     */
+    public function __construct($authenticatorID) {
+        parent::__construct($authenticatorID);
+    }
+
+    /**
+     * Getter of isTrusted.
+     */
+    public final function isTrusted() {
+        return $this->isTrusted;
+    }
+
+    /**
+     * Setter of isTrusted.
+     *
+     * @param bool $isTrusted
+     */
+    protected function setTrusted($isTrusted) {
+        $this->isTrusted = $isTrusted;
```

Comment:
It might not be a bad idea to force this to a boolean value or test it and throw an exception if it isn't.

Reply:
Everything can be casted to a boolean so it is not a problem. 

The doc says that it wants a boolean so if you pass it something else you are playing with fire.